src/databricks/labs/ucx/installer/workflows.py

```diff
@@ -525,12 +525,19 @@ def _deploy_workflow(self, step_name: str, settings):
         return None
 
     def _upload_wheel(self):
+        wheel_paths = []
         with self._wheels:
-            return self._wheels.upload_to_wsfs()
-
-    def _upload_wheel_runner(self, remote_wheel: str):
+            if self._config.upload_dependencies:
+                wheel_paths = self._wheels.upload_wheel_dependencies(
+                    ["databricks_sdk", "sqlglot"]
+                )  # TODO: add more dependencies if required
+            wheel_paths.append(self._wheels.upload_to_wsfs())
+            return wheel_paths
+
+    def _upload_wheel_runner(self, remote_wheel: list[str]):
         # TODO: we have to be doing this workaround until ES-897453 is solved in the platform
-        code = TEST_RUNNER_NOTEBOOK.format(remote_wheel=remote_wheel, config_file=self._config_file).encode("utf8")
+        remote_wheels_str = ", ".join([f"/Workspace{wheel}" for wheel in remote_wheel])
```

Comment:
does that work? write a working integration test to prove

Reply:
added int test